pandas/tests/generic/test_generic.py

```diff
@@ -910,3 +910,21 @@ def test_axis_classmethods(self, box):
             assert obj._get_axis_number(v) == box._get_axis_number(v)
             assert obj._get_axis_name(v) == box._get_axis_name(v)
             assert obj._get_block_manager_axis(v) == box._get_block_manager_axis(v)
+
+    def test_cache_on_copy(self):
```

Comment:
this should go in tests/frame/methods/test_copy.py

can you make sure the stuff being done below is the _minimal_ amount of stuff needed to trigger the bug in the status quo

Reply:
I checked and it seems the minimal amount is being done to trigger the bug. The reason df["y"] = [0] is added is because it causes df["a"].values[0] to be incorrect in addition to df when the cache is not invalidated on copy.

Reply:
@jbrockmendel 

> this should go in tests/frame/methods/test_copy.py

yes we should do that but as a followon (to capture all _copy) tests.